src/posit/connect/resources.py

```diff
@@ -0,0 +1,31 @@
+from abc import ABC, abstractmethod
+from typing import Generic, List, Optional, TypeVar
+
+
+T = TypeVar("T")
+
+
+class Resources(ABC, Generic[T]):
+    @abstractmethod
+    def create(self, *args, **kwargs) -> T:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def delete(self, *args, **kwargs) -> None:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def find(self, *args, **kwargs) -> List[T]:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def find_one(self, *args, **kwargs) -> Optional[T]:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def get(self, *args, **kwargs) -> T:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def update(self, *args, **kwargs) -> T:
```

Comment:
Same question as delete: would this be for bulk updates (regardless of whether the APIs would force us to do one request per entity)?